lib/DDG/Spice/Currency.pm

```diff
@@ -124,15 +125,18 @@ handle query_lc => sub {
 
         my $styler = number_style_for($amount);
         return unless $styler;
-
-        if ($cardinal ne '') {
+        
+        # only convert $amount if exists
+        if ($cardinal ne '' && $amount ne '') { 
             $amount = $styler->for_computation($amount);
 
-            if ($cardinal eq 'hundred')  { $amount *= 100 }
-            elsif ($cardinal =~ /(thousand|k)/i) { $amount *= 1000 }
-            elsif ($cardinal =~ /(million|m)/i)  { $amount *= 1000000 }
-            elsif ($cardinal =~ /(billion|b)/i)  { $amount *= 1000000000 }
-            elsif ($cardinal =~ /(trillion|t)/i) { $amount *= 1000000000000 }
+            if ($cardinal =~ /(hundred )/i)  { $amount *= 100 }
+            elsif ($cardinal =~ /(thousand |k )/i) { $amount *= 1000 }
+            elsif ($cardinal =~ /(million |m )/i)  { $amount *= 1000000 }
+            elsif ($cardinal =~ /(billion |b )/i)  { $amount *= 1000000000 }
+            elsif ($cardinal =~ /(trillion |t )/i) { $amount *= 1000000000000 }
```

Comment:
Can we format these with underscores (make them easier to read); e.g., `1_000_000`.